From 08a4ea27ab330b667e6d90a43f1bc8d82b6b6252 Mon Sep 17 00:00:00 2001
From: Song Bing <bing.song@nxp.com>
Date: Fri, 29 Jan 2016 15:21:30 +0800
Subject: [PATCH 2/2] Don't free texture when video playback.

Don't free texture when video playback. This can avoid video freeze
if user operate UI when video playback.

Upstream-Status: Inappropriate [i.MX specific]

Signed-off-by: Song Bing <bing.song@nxp.com>
---
 xbmc/Application.cpp |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/xbmc/Application.cpp b/xbmc/Application.cpp
index 5425322..0a866a9 100644
--- a/xbmc/Application.cpp
+++ b/xbmc/Application.cpp
@@ -5170,9 +5170,11 @@ void CApplication::ProcessSlow()
 #endif
 
   if (!m_pPlayer->IsPlayingVideo())
+  {
     g_largeTextureManager.CleanupUnusedImages();
+    g_TextureManager.FreeUnusedTextures(5000);
+  }
 
-  g_TextureManager.FreeUnusedTextures(5000);
 
 #ifdef HAS_DVD_DRIVE
   // checks whats in the DVD drive and tries to autostart the content (xbox games, dvd, cdda, avi files...)
-- 
1.7.9

